api/src/services/__tests__/programsService.ts

```diff
@@ -611,4 +623,26 @@ describe('programsService', () => {
       });
     });
   });
+
+  describe('listParticipantActivitiesCompletionForProgram', () => {
+    const programId = 1;
+    const principalId = 3;
+    it('should return a list of participant activities with their completion statuses', async () => {
+      mockQuery(
+        'select `activity_id`, `completed` from `participant_activities` where `program_id` = ? and `principal_id` = ?',
+        [programId, principalId],
+        [
+          { activity_id: 7, completed: false },
```

Comment:
Perhaps there's a way to extract this information from `participantActivitiesForProgram` (declared on line 174) instead of repeating their declaration here?